domain/src/test/java/org/oppia/domain/oppialogger/exceptions/ExceptionsControllerTest.kt

```diff
@@ -0,0 +1,350 @@
+      .onChanged(oppiaExceptionLogsResultCaptor.capture())
+
+    val exceptionLog = oppiaExceptionLogsResultCaptor.value.getOrThrow().getExceptionLog(0)
+    val exception = exceptionLogToException(exceptionLog)
+    assertThat(exception.message).isEqualTo(exceptionThrown.message)
+    assertThat(exception.stackTrace).isEqualTo(exceptionThrown.stackTrace)
+    assertThat(exception.cause?.message).isEqualTo(exceptionThrown.cause?.message)
+    assertThat(exception.cause?.stackTrace).isEqualTo(exceptionThrown.cause?.stackTrace)
+    assertThat(exceptionLog.exceptionType).isEqualTo(ExceptionType.FATAL)
+  }
+
+  @ExperimentalCoroutinesApi
+  @InternalCoroutinesApi
+  @Test
+  fun testController_logExceptions_exceedLimit_checkCorrectEviction() {
+    networkConnectionUtil.setCurrentConnectionStatus(NetworkConnectionUtil.ConnectionStatus.NONE)
+
+    exceptionsController.logFatalException(Exception("TEST1", Throwable("ONE")), 1556094120000)
+    exceptionsController.logNonFatalException(Exception("TEST2", Throwable("TWO")), 1556094110000)
+    exceptionsController.logFatalException(Exception("TEST3", Throwable("THREE")), 1556094100000)
+    exceptionsController.logFatalException(Exception("TEST4", Throwable("FOUR")), 1556094000000)
+
+    val cachedExceptions = exceptionsController.getExceptionLogs()
+    cachedExceptions.observeForever(mockOppiaExceptionLogsObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+
+    verify(mockOppiaExceptionLogsObserver, atLeastOnce())
+      .onChanged(oppiaExceptionLogsResultCaptor.capture())
+
+    val exceptionOne = oppiaExceptionLogsResultCaptor.value.getOrThrow().getExceptionLog(0)
+    val exceptionTwo = oppiaExceptionLogsResultCaptor.value.getOrThrow().getExceptionLog(1)
+    val cacheStoreSize = oppiaExceptionLogsResultCaptor.value.getOrThrow().exceptionLogList.size
+
+    assertThat(cacheStoreSize).isEqualTo(2)
+    // In this case, 3 FATAL and 1 NON-FATAL exception was logged. So while pruning, none of the retained logs should have NON-FATAL exception type.
+    assertThat(exceptionOne.exceptionType).isNotEqualTo(ExceptionType.NON_FATAL)
+    assertThat(exceptionTwo.exceptionType).isNotEqualTo(ExceptionType.NON_FATAL)
+    // If we analyse the order of logging of exceptions, we can see that record pruning will begin from the logging of the third record.
+    // At first, the second exception log will be removed as it has NON-FATAL exception type and the exception logged at the third place will become the exception record at the second place in the store.
+    // When the forth exception gets logged then the pruning will be purely based on timestamp of the exception as both exception logs have FATAL exception type.
+    // As the third exceptions's timestamp was lesser than that of the first event, it will be pruned from the store and the forth exception will become the second exception in the store.
+    assertThat(exceptionOne.timestampInMillis).isEqualTo(1556094120000)
```

Comment:
Please save these & other timestamps to distinct constants to help provide quick cross-reference checking for methods like this.

Reply:
I kept them as numbers because it would've been easier to cross-verify pruning. Changed to constants now.